dali/c_api/c_api.cc

```diff
@@ -546,7 +546,8 @@ void daliDeletePipeline(daliPipelineHandle* pipe_handle) {
   auto *bs_map = reinterpret_cast<batch_size_map_t *>(pipe_handle->batch_size_map);
   DALI_ENFORCE(pipeline != nullptr && ws != nullptr, "Pipeline already deleted");
   if (pipe_handle->copy_stream) {
-    CUDA_CALL(cudaStreamDestroy(pipe_handle->copy_stream));
+    CUDA_CALL(cudaStreamSynchronize(pipe_handle->copy_stream));
```

Comment:
Why you need to synchronize before putting it back to the pool?

Reply:
I don't think I really need to. I think it's a historical artifact - initially, I destroyed the stream, but then the stream-ordered deletion failed with hard errors (the stream handle was still used) - that's why I added the synchronization. Subsequently, I realized that it's not possible to use stream-aware deleters _and_ destroy streams. That's why the stream pool introduced. I changed the stream destruction to putting it back in the pool, but I haven't removed the synchronization added before.
Having said that, I don't think it's good to put a stream with pending work back in the pool - especially if the stream pool uses LIFO order.

Reply:
Does it make sense to add synchronization in the `Put` function?

Reply:
Not really. `cudaStreamDestroy` doesn't synchronize - the stream will be kept alive until it drains. I think we should switch to FIFO or do some magic with callbacks/host functions.

Reply:
I guess this is ToDo for a next PR, right?

Reply:
Definitely.